src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -138,6 +149,27 @@ private static String retrieveText(ByteString imageBytes)
     return rawText;
   }
 
+  /** Generates categories for the provided text. */
+  private static Set<String> categorizeText(String text) throws IOException {
+    Set<String> categories = Collections.emptySet();
+
+    try (LanguageServiceClient client = LanguageServiceClient.create()) {
+      Document document = Document.newBuilder().setContent(text).setType(Type.PLAIN_TEXT).build();
+      ClassifyTextRequest request = ClassifyTextRequest.newBuilder().setDocument(document).build();
+
+      // TODO: Check if ApiException was thrown
+      ClassifyTextResponse response = client.classifyText(request);
+
+      // TODO: Parse category strings into more natural categories
+      categories = response.getCategoriesList()
+                       .stream()
+                       .map(category -> category.getName())
+                       .collect(Collectors.toSet());
```

Comment:
I saw Basaam mentioned you should be using an immutable set, so here it would be .collect(ImmutableSet.toImmutableSet())